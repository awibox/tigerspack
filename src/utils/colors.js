export const colors = {
  primary: {
    color: '#3F88C5',
    text: '#fff',
    hoverColor: '#619DCF',
    hoverText: '#fff',
  },
  yellow: {
    color: '#FDE74C',
    text: '#304355',
    hoverColor: '#FDEF8D',
    hoverText: '#596977',
  },
  danger: {
    color: '#D15130',
    text: '#fff',
    hoverColor: '#E76846',
    hoverText: '#fff',
  },
  warning: {
    color: '#FA7921',
    text: '#fff',
    hoverColor: '#FB9D5D',
    hoverText: '#fff',
  },
  success: {
    color: '#9BC53D',
    text: '#fff',
    hoverColor: '#B6D471',
    hoverText: '#fff',
  },
  info: {
    color: '#5BC0EB',
    text: '#fff',
    hoverColor: '#87D1F0',
    hoverText: '#fff',
  },
  dark: {
    color: '#304355',
    text: '#fff',
    hoverColor: '#596977',
    hoverText: '#fff',
  },
  light: {
    color: '#E5E8EC',
    text: '#304355',
    hoverColor: '#EEF0F2',
    hoverText: '#596977',
  },
};

export const colorsArray = ['primary', 'danger', 'success'];
