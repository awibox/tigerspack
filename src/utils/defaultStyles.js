export const defaultStyles = {
  animation: 'all .4s ease',
  indent: 15,
  buttonSize: 3,
  borderRadius: 3,
  mediumFontSize: '14px',
};

export const setDefaultStyles = (newStyles) => {
  Object.keys(newStyles).forEach((style) => {
    defaultStyles[style] = newStyles[style];
  });
};
