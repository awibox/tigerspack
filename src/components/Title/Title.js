import React from 'react';
import PropTypes from 'prop-types';
import { colors } from '../../utils/colors';

const Title = (props) => {
  const {
    children,
    indent,
    size,
    padding,
    border,
    ...otherProps
  } = props;

  const titleStyles = {
    fontWeight: 500,
    padding: `${padding}px`,
    marginBottom: `${indent}px`,
    borderBottom: border && `1px solid ${colors[border].color}`,
    paddingBottom: border && `${indent}px`,
  };

  switch (size) {
    case 'large':
      return <h1 css={titleStyles} {...otherProps}>{children}</h1>;
    case 'small':
      return <h3 css={titleStyles} {...otherProps}>{children}</h3>;
    default:
      return <h2 css={titleStyles} {...otherProps}>{children}</h2>;
  }
};

Title.propTypes = {
  children: PropTypes.any.isRequired,
  indent: PropTypes.number,
  size: PropTypes.oneOf(['small', 'medium', 'large']),
  padding: PropTypes.number,
  border: PropTypes.string,
};

Title.defaultProps = {
  padding: 0,
  indent: 10,
};

export default Title;
