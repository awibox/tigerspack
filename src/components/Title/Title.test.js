import React from 'react';
import { shallow } from 'enzyme';
import Title from './Title';
import { colors } from '../../utils/colors';

const testValue = 1;
const testColor = 'primary';

describe('Title', () => {
  it('Default Title', () => {
    const component = shallow(<Title>Test Blank</Title>);
    expect(component).toMatchSnapshot();
    expect(component.html().substr(1, 2)).toEqual('h2');
  });
  it('Title size', () => {
    const component = shallow(<Title size={'large'}>Test Blank</Title>);
    expect(component.html().substr(1, 2)).toEqual('h1');
  });
  it('Title small', () => {
    const component = shallow(<Title size={'small'}>Test Blank</Title>);
    expect(component.html().substr(1, 2)).toEqual('h3');
  });
  it('Title indent', () => {
    const component = shallow(<Title indent={testValue}>Test Blank</Title>);
    expect(component.get(0).props.css.marginBottom).toEqual(`${testValue}px`);
  });
  it('Title padding', () => {
    const component = shallow(<Title padding={testValue}>Test Blank</Title>);
    expect(component.get(0).props.css.padding).toEqual(`${testValue}px`);
  });
  it('Title border', () => {
    const component = shallow(<Title border={testColor}>Test Blank</Title>);
    expect(component.get(0).props.css.borderBottom).toEqual(`1px solid ${colors[testColor].color}`);
  });
});
