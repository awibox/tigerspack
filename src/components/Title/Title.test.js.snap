// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title Default Title 1`] = `
<EmotionCssPropInternal
  __EMOTION_LABEL_PLEASE_DO_NOT_USE__="Component"
  __EMOTION_TYPE_PLEASE_DO_NOT_USE__="h2"
  css={
    Object {
      "borderBottom": "",
      "fontWeight": 500,
      "marginBottom": "10px",
      "padding": "0px",
      "paddingBottom": "",
    }
  }
>
  Test Blank
</EmotionCssPropInternal>
`;
